src/codeintel/lib/codeintel2/database/langlib.py

```diff
@@ -1342,11 +1342,18 @@ def cull_mem(self):
             now = time.time()
             for dbsubpath, (index, atime) \
                     in self._index_and_atime_from_dbsubpath.items():
+                # Don't delete if recently accessed
                 if now - atime > TIME_SINCE_ACCESS:
                     if dbsubpath in self._is_index_dirty_from_dbsubpath:
                         self.save_index(dbsubpath, index)
                         del self._is_index_dirty_from_dbsubpath[dbsubpath]
                     del self._index_and_atime_from_dbsubpath[dbsubpath]
+                # It was recently accessed
+                else:
+                    # If it's dirty, write it to disk
+                    if dbsubpath in self._is_index_dirty_from_dbsubpath:
+                        self.save_index(dbsubpath, index)
+                        del self._is_index_dirty_from_dbsubpath[dbsubpath]
```

Comment:
This completely bypasses the previous conditional. I imagine doing this will land you in race conditions or worse. Point being that the conditional is there for a reason.

Reply:
I can remove this if you want.  It just seemed like an index that is frequently used might not be written to disk.  I wasn't sure if that would cause issues on the next restart.

Reply:
Indexes aren't permanent one way or another, so even if it were stored it would still come across it again and invalidate if necessary. 

I'd rather deal with that unknown than the unknown impact of removing the timer that was clearly added there for a reason. :)

Reply:
Ok.  I removed it.  I wasn't sure if it was intentional or not.  I thought maybe it was supposed to be written, but not removed from the cache.